drivers/volume/portworx/portworx.go

```diff
@@ -328,6 +328,62 @@ func (d *portworx) isMetadataNode(node node.Node, address string) (bool, error)
 	return false, nil
 }
 
+func (d *portworx) CloneVolume(volumeID string) (string, error) {
+	volDriver := d.getVolDriver()
+	volumeInspectResponse, err := volDriver.Inspect(d.getContext(), &api.SdkVolumeInspectRequest{VolumeId: volumeID})
+	if err != nil {
+		return "", fmt.Errorf("failed to find volume %v due to %v", volumeID, err)
+	}
+	pxVolume := volumeInspectResponse.Volume
+
+	volID := pxVolume.Id
+	cloneVolumeName := pxVolume.Locator.Name + "_clone"
+
+	volumeCloneResp, err := volDriver.Clone(d.getContext(), &api.SdkVolumeCloneRequest{ParentId: volID, Name: cloneVolumeName})
+	if err != nil {
+		logrus.Infof("Error %v", err)
```

Comment:
you can remove this log statement as the error is getting printed in next line

Reply:
Removed.